src/ApplicationManager.cpp

```
@@ -332,18 +338,27 @@ void ApplicationManager::handleSpoolDetected(const AppMessage& msg) {
-        lcdManager->updateScreen("**** Spool ****", "*** Scanned ***", line1, line2);
-    } else if (lcdManager) {
+        const auto& s = msg.payload.spoolDetected;
+        DisplaySpoolData spool{};
+        strncpy(spool.brand, s.manufacturer, sizeof(spool.brand) - 1);
+        strncpy(spool.material, s.material_name, sizeof(spool.material) - 1);
+        snprintf(spool.colorHex, sizeof(spool.colorHex), "%02X%02X%02X",
+                 s.primary_color[0], s.primary_color[1], s.primary_color[2]);
+        spool.remainingWeight = s.kg_remaining * 1000.0f;
+        spool.totalWeight = s.initial_weight_g;
+        // Map tag format string to tag type constant
+        if (strcmp(s.tag_format, "OpenPrintTag") == 0) spool.tagType = 1;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)